hypershift-operator/controllers/hostedcluster/internal/platform/gcp/gcp.go

```diff
@@ -14,75 +14,438 @@ package gcp
+		Port: apiEndpoint.Port,
+	}
+
+	// Note: CAPG will handle failure domains and status automatically based on region
+	// We don't modify Status fields directly as that's managed by CAPG controllers
+
+	return nil
+}
+
+// CAPIProviderDeploymentSpec implements CAPG controller deployment specification.
+// This method creates a deployment spec for the CAPG (Cluster API Provider GCP)
+// controller with proper image handling, feature gates, and WIF preparation.
+func (p GCP) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
+	// Image priority: annotation > env var > payload
+	providerImage := p.capiProviderImage
+	if envImage := os.Getenv(images.GCPCAPIProviderEnvVar); len(envImage) > 0 {
+		providerImage = envImage
+	}
+	if override, ok := hcluster.Annotations[hyperv1.ClusterAPIGCPProviderImage]; ok {
+		providerImage = override
+	}
+
+	// Validate image is available
+	if providerImage == "" {
+		return nil, fmt.Errorf("no GCP CAPI provider image specified")
+	}
+
+	// GCP-specific feature gates based on payload version
+	featureGates := []string{
+		"MachinePool=false", // Disable for Phase 1
+	}
+
+	// Version-conditional feature gates (future-proofing)
+	if p.payloadVersion != nil && p.payloadVersion.Major == 4 && p.payloadVersion.Minor > 16 {
+		featureGates = append(featureGates, "ClusterResourceSet=false") // Example
+	}
+
+	args := []string{
+		"--namespace=$(MY_NAMESPACE)",
+		"--leader-elect=true",
+		fmt.Sprintf("--feature-gates=%s", strings.Join(featureGates, ",")),
+		"--v=2",
+	}
+
+	containers := []corev1.Container{
+		{
+			Name:            "manager",
+			Image:           providerImage,
+			ImagePullPolicy: corev1.PullIfNotPresent, // AWS/Azure pattern
+			Args:            args,
+			Env: []corev1.EnvVar{
+				{Name: "MY_NAMESPACE", ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"}}},
+				{Name: "GOOGLE_APPLICATION_CREDENTIALS", Value: "/home/.gcp/application_default_credentials.json"},
+				{Name: "GOOGLE_CLOUD_PROJECT", Value: hcluster.Spec.Platform.GCP.Project},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "credentials",
+					MountPath: "/home/.gcp",
+					ReadOnly:  true,
+				},
+				{
+					Name:      "capi-webhooks-tls",
+					ReadOnly:  true,
+					MountPath: "/tmp/k8s-webhook-server/serving-certs",
+				},
+				{
+					Name:      "token",
+					MountPath: "/var/run/secrets/openshift/serviceaccount",
+				},
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("10m"),
+					corev1.ResourceMemory: resource.MustParse("128Mi"),
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: ptr.To(false),
+				Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
+				RunAsNonRoot:             ptr.To(true),
+			},
```

Comment:
This will be done automatically when https://github.com/openshift/hypershift/pull/7219 is merged